drivers/timer/xtensa_sys_timer.c

```diff
@@ -465,6 +479,18 @@ void _xt_tick_divisor_init(void)
 void _timer_int_handler(void *params)
 {
 	ARG_UNUSED(params);
+
+#ifdef CONFIG_XTENSA_ASM2
+	/* FIXME: the legacy xtensa code did this in the assembly
+	 * hook, and was a little more sophisticated.  We should track
+	 * the delta from the last set time, not the current time.
+	 * And the earlier code was even prepared to handle missed
+	 * ticks by calling the handler function multiple times if the
+	 * delta was more than one _xt_tick_divisor.
+	 */
+	SET_TIMER_FIRE_TIME(GET_TIMER_CURRENT_TIME() + _xt_tick_divisor);
```

Comment:
I had a fever dream that one day this driver was fixed up to use IRQ_CONNECT() and all the custom asm stuff for it was nuked from orbit. I should file a bug on that.